### DISCUSSION TITLE
Generating custom banner in a library

### DISCUSSION THREAD

**Discussion by triceo** [quarkusio/quarkus#36957] [Q&A]
Hello there,

I am trying to replace the Quarkus banner with my own. I am doing it using code like this:

    @BuildStep
    void produceBanner(BuildProducer<GeneratedResourceBuildItem> resourceProducer) {
        var banner = "My banner";
        var build_item =
                new GeneratedResourceBuildItem("default_banner.txt", banner.getBytes(StandardCharsets.UTF_8));
        resourceProducer.produce(build_item);
    }

This tries to put the `default_banner.txt` into the JAR that Quarkus generates. My reason for doing this is that I need to generate the banner dynamically, and I am a Quarkus extension - people will be using the extension, and I want them to see this banner.

However, the banner is not put anywhere in the JAR that I can see, and when the app using my extension starts, it shows the default Quarkus banner. This tells me I'm on the wrong path.

What can I do to make sure the users of my extension see my generated banner?

**Comment by ramiswailem** [comment]
Or use "quarkus.banner.path"

**Reply by triceo** [reply]
If I were an end-user, that would be straightforward.
But how to do what when I am an extension?
